### PR TITLE
[Simulator] Fix undraggable rect & line

### LINKED ISSUE #1091: [Simulator] Disabled price manipulation
https://github.com/bancorprotocol/carbon-app/assets/8143464/83068d57-dbb9-4dcf-b27d-450a95a41a1a

### PR DESCRIPTION
**Pull request by GrandSchtroumpf**
fix #1091 
Wait for the svg element to be rendered before running selection

### CODE REVIEW

**Review by ashachaf — APPROVED** [review]

File: src/components/simulator/input/d3Chart/utils.ts [hunk 1]
```diff
@@ -1,9 +1,20 @@
...
 import { CandlestickData } from 'libs/d3/types';
+import { useEffect, useState } from 'react';
 
 export const getSelector = (selector: string) => select(`.${selector}`);
 
+export const useSelectable = (selector: string) => {
+  const [isSelectable, setIsSelectable] = useState(false);
+  useEffect(() => {
+    const selection = getSelector(selector);
+    if (!selection.size()) return;
```
Comment (zavelevsky) [inline src/components/simulator/input/d3Chart/utils.ts:12]: question: shouldn't there be a code path that sets it to false? e.g. when the selection size is 0?

Comment (GrandSchtroumpf) [reply src/components/simulator/input/d3Chart/utils.ts:12]: The selection size is 0 when D3 doesn't find the SVGElement. It's a way to wait for the element to be rendered in the DOM.

**Review by pingustar — APPROVED** [review]
LGTM

**Review by zavelevsky — APPROVED** [review]
LGTM

**Pull request merged**